### PR TITLE
More efficient indexing by CartesianIndices

### PR DESCRIPTION
**Pull request by mbauman**
Treat CartesianIndices as its tuple of ranges when indexing. This allows views that use CartesianIndices as indices to be strided and changes the complexity of boundschecking from O(n^d) to O(d)

### CODE REVIEW

**Review by eulerkochy — CHANGES_REQUESTED** [review]
`@testset "indexing by CartesianIndices"` should be `@testset "indexing by CartesianIndices" begin`, imo

File: test/arrayops.jl [hunk 1]
```diff
@@ -2622,3 +2622,17 @@ end
 # Fix oneunit bug for unitful arrays
 @test oneunit([Second(1) Second(2); Second(3) Second(4)]) == [Second(1) Second(0); Second(0) Second(1)]
 
+@testset "indexing by CartesianIndices"
```
Comment (mbauman) [inline test/arrayops.jl]:
```suggestion
@testset "indexing by CartesianIndices" begin
```

**Comment by mbauman** [comment]
Thanks @eulerkochy!

**Review by timholy — APPROVED** [review]
Arrived while I was immersed in other tasks (writing JuliaIntepreter, I think). Nice!

**Comment by KristofferC** [comment]
Rebased it to re-run CI.

**Pull request merged**